app/components/post_draft/post_draft.js

```diff
@@ -375,14 +476,34 @@ export default class PostDraft extends PureComponent {
 
     sendMessage = () => {
         const value = this.input.current?.getValue() || '';
-        const {enableConfirmNotificationsToChannel, membersCount, useChannelMentions} = this.props;
+        const {enableConfirmNotificationsToChannel, membersCount, useGroupMentions, useChannelMentions, channelMemberCountsByGroup} = this.props;
         const notificationsToChannel = enableConfirmNotificationsToChannel && useChannelMentions;
+        const notificationsToGroups = enableConfirmNotificationsToChannel && useGroupMentions;
         const toAllOrChannel = this.textContainsAtAllAtChannel(value);
+        const groupMentions = notificationsToGroups ? this.groupsMentionedInText(value) : [];
```

Comment:
in line 489 you are checking that is does not have an `AtAllOrChannel` mentions, then why would you run the regex to see if there any matching group mentions here? would be better to do
```suggestion
        const groupMentions = (!toAllOrChannel && notificationsToGroups) ? this.groupsMentionedInText(value) : [];
```

Reply:
Good call!